src/components/recommender/extension-recommender.ts

```diff
@@ -0,0 +1,66 @@
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { Kubectl } from '../../kubectl';
+
+interface Recommendations {
+    [crdApi: string]: string[];
+}
+
+async function recommendExtensionsToUser(tags: string[]): Promise<void> {
+    if (tags.length > 0) { // do not prompt if we cannot recommend anything
+        const answer = await vscode.window.showInformationMessage('Do you want to search for extensions based on CRDs that exist on your kubernetes cluster?', 'Don\'t ask again', 'Ignore', 'Search for Extensions');
+        if (answer === 'Search for Extensions') {
+            const searchValue = tags.map((it) => `@tag:${it}`).join(' ');
+            vscode.commands.executeCommand('workbench.extensions.search', searchValue);
+        } else if (answer === 'Don\'t ask again') {
+            await vscode.workspace.getConfiguration('extensions').update('ignoreRecommendations', true);
+        }
+    }
+}
+
+async function getCRDApiGroups(kubectl: Kubectl): Promise<string[]> {
+    const result = new Set<string>();
+    const invokeResult = await kubectl.invokeCommand('api-versions');
+    if (invokeResult.resultKind === 'exec-succeeded') {
+        try {
+            const res = invokeResult.stdout.split('\n');
+            res.forEach((i: any) => result.add(i));
```

Comment:
There is no need of having this i guess, split already returns an array and api-versions doesn't have duplicate (right?) so no need to use a set. We should add a verification that there are no empty values (eg last '\n'). 
Something like 
```
try {
 return invoke.....split('\n').filter(s => s not empty)
} catch .... {}
return [];
```